src/plugins/dashboard/public/dashboard_app/_dashboard_app_strings.ts

```diff
@@ -154,6 +154,26 @@ export const shareModalStrings = {
       defaultMessage:
         'One or more panels on this dashboard have changed. Before you generate a snapshot, save the dashboard.',
     }),
+  getEmbedShareWarning: () =>
+    i18n.translate('dashboard.embedShare.draftWarning', {
+      defaultMessage:
+        'This dashboard has unsaved changes. Consider saving your dashboard before copying the embed link.',
+    }),
+  getDraftSharePanelChangesWarning: (hasSessionStorageEnabled: boolean) =>
+    hasSessionStorageEnabled
+      ? i18n.translate('dashboard.snapshotShare.panelChangesWarning', {
+          defaultMessage:
+            'One or more panels on this dashboard have changed. Before generating a link, save the dashboard.',
+        })
+      : i18n.translate('dashboard.snapshotShare.panelChangesWarningNoSessionStorage', {
+          defaultMessage:
+            'One or more panels on this dashboard have changed, resolving the link for this draft state requires session storage to be enabled, consider saving changes to the dashboard.',
```

Comment:
```suggestion
            'One or more panels on this dashboard have changed, resolving the link for this draft state requires session storage to be enabled, consider saving changes to the dashboard.',
```

This is confusing to me. I'm not sure we need to be that specific, but defaulting to the opinion of @florent-leborgne or @wajihaparvez here 🤔

Reply:
```suggestion
            'You are about to share a dashboard with unsaved changes, and the link may not work properly. Save the dashboard first to create a permanent link.',
```
What about something more generic like this?

Reply:
This sounds perfect, thanks @florent-leborgne!